src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -1232,6 +1242,10 @@ protected FormValidation doTestConnection(@AncestorInPath ItemGroup context, URL
                                         .error("The EC2 key pair private key isn't registered to this EC2 region (fingerprint is "
                                                 + pk.getFingerprint() + ")");
                         }
+                    } else {
+                        LOGGER.warning(() -> "SSH credential was invalid");
+                        return FormValidation.error("SSH Credential was invalid");
```

Comment:
“Invalid” how? I think you mean that no credentials of that id exist. Not something you would normally see since the dropdown should only be letting you pick valid ids.